inst/pages/exercises.qmd

```diff
@@ -1741,6 +1741,175 @@ pheatmap(
     annotation_col = as.data.frame(sample_metadata))
 ```
 
+### *bonus* Regression charts to visualize microbiome data
+
+Regression analysis is a powerful statistical method that can help examine
+the relationship between two or more variables. In microbiome research,
+regression charts help in visualizing how changes in microbial communities
+correlate with various environmental factors, health outcomes, or other
+biological variables.
+
+Let us do a regression analysis exercise using the `GlobalPatterns` dataset
+against the Shannon index ("ShannonIndex"), a common measure of alpha diversity,
+which quantifies the diversity within each sample. This index will serve as the
+dependent variable in our regression model examples.
+
+We extract SampleType as a categorical(discrete) variable and coverage as a
+continuous variable from the dataset’s metadata. These variables will be used
+as predictors/independent variables in our linear regression models.
+
+Steps:
+1. Import the dataset into R; here we are using a microbiome dataset.
+2. Calculate Alpha Diversity (Shannon Index), i.e. the dependent variable and 
+define the covariates, the independent variable.
+3. Fit linear model using discrete variable "SampleType", where Shannon diversity
+is the response variable, and SampleType is the predictor. 
+4. Visualize the results using a boxplot. This plot helps in understanding the
+distribution of diversity across discrete sample types.
+5. Fit linear model using continuous variable "coverage", where Shannon diversity 
+is the response variable, and coverage is the predictor.  
+6. Visualize the model with a scatterplot. In R, you can also use 
+[`ggplot2`](https://ggplot2.tidyverse.org/) for
+plotting and `lm()` function for linear modeling.
+7. Examine the slope, intercept, and R-squared value to understand the 
+relationship between the variables.
+8. Summarize results with `gtsummary`. The [`gtsummary`]
+(https://www.danieldsjoberg.com/gtsummary/) package is used to generate a summary 
+table of the regression model, which is easy to interpret and present in reports
+or publications.
+9. Visualize coefficients with `jtools`. The [`jtools`]
+(https://cran.r-project.org/web/packages/jtools/index.html) package provides a visual
+representation of the model’s coefficients, including their confidence intervals, allowing
+for a clear understanding of the effect sizes.
+10. Finally, interpret the analysis outcomes. 
+
+Visualization methods from packages like 
```

Comment:
Better to show the code step-by-step?

Reply:
I have added relevant comments in the code to show the steps too, do you think I should merge the steps into comments?

Reply:
Plus, this gives some space in between the code to talk about the interpretations.

Reply:
Yes, split the code into multiple chunks and explain more per each step.